src/Composer/Installer.php

```diff
@@ -0,0 +1,209 @@
+    }
+
+    /**
+     * Returns the active plugin instances of a given type (see class constants).
+     *
+     * @param string $type
+     * @param bool   $reverseOrder
+     *
+     * @return array
+     */
+    public function getInstancesOf($type, $reverseOrder = false)
+    {
+        $plugins = array_filter(
+            $this->getInstances(),
+            function ($plugin) use ($type) {
+                return is_a($plugin, $type);
+            }
+        );
+
+        $plugins = $reverseOrder ? array_reverse($plugins, true) : $plugins;
+
+        return $plugins;
+    }
+}
+
+PHP;
+
+    /**
+     * Sets the Contao Manager plugins.
+     *
+     * @param Locker               $locker
+     * @param RootPackageInterface $rootPackage
+     *
+     * @throws \RuntimeException
+     */
+    public function dumpPlugins(Locker $locker, RootPackageInterface $rootPackage)
+    {
+        $plugins = [];
+
+        $disabled = $this->getDisabledPackages($rootPackage);
+        $lockData = $locker->getLockData();
+        $lockData['packages-dev'] = isset($lockData['packages-dev']) ? $lockData['packages-dev'] : [];
+
+        foreach (array_merge($lockData['packages'], $lockData['packages-dev']) as $package) {
+            if (in_array($package['name'], $disabled, true)) {
+                continue;
+            }
+
+            if (isset($package['extra']['contao-manager-plugin'])) {
+                if (!class_exists($package['extra']['contao-manager-plugin'])) {
+                    throw new \RuntimeException(
+                        sprintf('Plugin class "%s" not found', $package['extra']['contao-manager-plugin'])
+                    );
+                }
+
+                $plugins[$package['name']] = new $package['extra']['contao-manager-plugin']();
+            }
+        }
+
+        $plugins = $this->orderPlugins($plugins);
+
+        // Instantiate a global plugin to load AppBundle or other customizations
+        $appPlugin = '\ContaoManagerPlugin';
+
+        if (class_exists($appPlugin)) {
+            $plugins['app'] = new $appPlugin();
+        }
+
+        $this->dumpClass($plugins);
+    }
+
+    /**
+     * Finds disabled packages in the root package extras.
+     *
+     * @param RootPackageInterface $rootPackage
+     *
+     * @return array
+     */
+    private function getDisabledPackages(RootPackageInterface $rootPackage)
+    {
+        $extra = $rootPackage->getExtra();
+
+        if (!isset($extra['contao-manager']['disabled-packages'])) {
+            return [];
+        }
+
+        return (array) $extra['contao-manager']['disabled-packages'];
+    }
+
+    /**
+     * Dumps the PluginLoader class.
+     */
+    private function dumpClass(array $plugins)
+    {
+        $content = sprintf(
+            static::$generatedClassTemplate,
+            'cla'.'ss '.'PluginLoader', // note: workaround for regex-based code parsers :-(
+            'unserialize('.var_export(serialize($plugins), true).')'
```

Comment:
Wouldn’t it be better for OP caching to use something like this instead of `serialize()`?

```php
$content = sprintf(
    static::$generatedClassTemplate,
    'cla'.'ss '.'PluginLoader', // note: workaround for regex-based code parsers :-(
    "[\n".implode("\n", array_map(function($plugin) {
        return 'new \\'.get_class($plugin).'(),';
    }, $plugins))."\n]"
);
```

Reply:
This seems weird to me, too. `ocramius/package-versions` does not use `unserialize()` either, does it?

Reply:
Yeah, weird to me too. I tried to use `var_export` but the namespaces were "broken" and it used the `__set_state` method which is not implemented in the bundle classes.

Reply:
@aschempp did you try my code snippet from above?

Reply:
I did not. But it would also change the current implementation (meaning that the PluginLoader "knows" how a plugin can be instantiated). Not sure if we want to change that?

Reply:
> (meaning that the PluginLoader "knows" how a plugin can be instantiated)

But that’s already the case, the installer instantiates the plugins on line 125 currently https://github.com/contao/manager-plugin/pull/9/files#diff-74425bc790838a69d0d16f06fc313edbR125